performanceplatform/utils/http_with_backoff.py

```diff
@@ -8,6 +9,26 @@
 _MAX_RETRIES = 5
 
 
+def parse_reason(response, content):
+    """
+    Google return a JSON document describing the error. We should parse this
+    and extract the error reason. See
+    https://developers.google.com/analytics/devguides/reporting/core/v3/coreErrors
+    """
+    def first_error(data):
+        errors = data['error']['errors']
+        if len(errors) > 1:
+            # we have more than one error. We should capture that?
+            logging.info('Received {} errors'.format(len(errors)))
+        return errors[0]
+
+    try:
+        json_data = json.loads(content)
+        return first_error(json_data)['reason']
+    except:
```

Comment:
Could make this explicitly catch JSON errors, in case the response changes and `first_error` starts failing.